Bit-level parsing seems broken in 6.0.0

## Problem
I'm having problems using the bit-level parses in `nom::bits`. The error message involves `expected an Fn<...> closure, found impl FnMut<...>`. If I recall correctly, one of the changes in 6.0.0 was moving to `FnMut` - maybe you forgot to update something in the `bits` module?

## Environment
- Rust version: 1.47.0
- nom version: 6.0.0

## Test case
```
use nom::{
    bits::{bits, complete::take},
    sequence::tuple,
    IResult
};

fn main() {
    println!("Hello, world!");
}

pub fn example(input: &[u8]) -> IResult<&[u8], (u8, u8)> {
    bits(tuple((
        take(1usize), take(1usize)
    )))(input)
}
```
This gives the following error message:
```
error[E0277]: expected a `std::ops::Fn<((_, usize),)>` closure, found `impl std::ops::FnMut<((_, usize),)>`
  --> src/main.rs:12:10
   |
12 |       bits(tuple((
   |  __________^
13 | |         take(1usize), take(1usize)
14 | |     )))(input)
   | |______^ expected an `Fn<((_, usize),)>` closure, found `impl std::ops::FnMut<((_, usize),)>`
   |
  ::: /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/nom-6.0.0/src/bits/mod.rs:41:6
   |
41 |     P: Fn((I, usize)) -> IResult<(I, usize), O, E1>,
   |        -------------------------------------------- required by this bound in `nom::bits`
   |
   = help: the trait `std::ops::Fn<((_, usize),)>` is not implemented for `impl std::ops::FnMut<((_, usize),)>`
```

## PS
Also I'm not sure I really understand how the bit-level parsing is supposed to work. I've wrapped my head around the "normal" parsing, but I don't get the return type of `nom::bits::complete::take` and how I'm supposed to use `nom::bits::bits` and `nom::bits::bytes`. More documentation and examples regarding that topic would be really helpful.

--- a/src/bits/mod.rs
+++ b/src/bits/mod.rs
@@ -34,11 +34,11 @@ use crate::traits::{ErrorConvert, Slice};
 /// assert_eq!(take_4_bits( sl ), Ok( (&sl[1..], 0xA) ));
 /// ```
 pub fn bits<I, O, E1: ParseError<(I, usize)> + ErrorConvert<E2>, E2: ParseError<I>, P>(
-  parser: P,
-) -> impl Fn(I) -> IResult<I, O, E2>
+  mut parser: P,
+) -> impl FnMut(I) -> IResult<I, O, E2>
 where
   I: Slice<RangeFrom<usize>>,
-  P: Fn((I, usize)) -> IResult<(I, usize), O, E1>,
+  P: FnMut((I, usize)) -> IResult<(I, usize), O, E1>,
 {
   move |input: I| match parser((input, 0)) {
     Ok(((rest, offset), res)) => {
@@ -58,7 +58,7 @@ pub fn bitsc<I, O, E1: ParseError<(I, usize)> + ErrorConvert<E2>, E2: ParseError
 ) -> IResult<I, O, E2>
 where
   I: Slice<RangeFrom<usize>>,
-  P: Fn((I, usize)) -> IResult<(I, usize), O, E1>,
+  P: FnMut((I, usize)) -> IResult<(I, usize), O, E1>,
 {
   bits(parser)(input)
 }
@@ -89,11 +89,11 @@ where
 /// assert_eq!(parse( input ), Ok(( &[][..], (0xd, 0xea, &[0xbe, 0xaf][..]) )));
 /// ```
 pub fn bytes<I, O, E1: ParseError<I> + ErrorConvert<E2>, E2: ParseError<(I, usize)>, P>(
-  parser: P,
-) -> impl Fn((I, usize)) -> IResult<(I, usize), O, E2>
+  mut parser: P,
+) -> impl FnMut((I, usize)) -> IResult<(I, usize), O, E2>
 where
   I: Slice<RangeFrom<usize>> + Clone,
-  P: Fn(I) -> IResult<I, O, E1>,
+  P: FnMut(I) -> IResult<I, O, E1>,
 {
   move |(input, offset): (I, usize)| {
     let inner = if offset % 8 != 0 {
@@ -122,7 +122,7 @@ pub fn bytesc<I, O, E1: ParseError<I> + ErrorConvert<E2>, E2: ParseError<(I, usi
 ) -> IResult<(I, usize), O, E2>
 where
   I: Slice<RangeFrom<usize>> + Clone,
-  P: Fn(I) -> IResult<I, O, E1>,
+  P: FnMut(I) -> IResult<I, O, E1>,
 {
   bytes(parser)(input)
 }

--- a/tests/issues.rs
+++ b/tests/issues.rs
@@ -347,3 +347,19 @@ named!(issue_962<&[u8], Vec<&[u8]>>,
         }
     )
 );
+
+#[test]
+fn issue_1231_bits_expect_fn_closure() {
+    use nom::bits::{bits, complete::take};
+    use nom::error::Error;
+    use nom::sequence::tuple;
+    pub fn example(input: &[u8]) -> IResult<&[u8], (u8, u8)> {
+        bits::<_, _, Error<_>, _, _>(tuple((
+                    take(1usize), take(1usize)
+        )))(input)
+    }
+    assert_eq!(
+        example(&[0xff]),
+        Ok((&b""[..], (1, 1)))
+    );
+}

